### ISSUE TITLE
Cache API throws a Not Found exception

### ISSUE THREAD

**Issue by oscarotero** [denoland/deno#19696]
I have the following code that worked fine until now:
```js
const cached = await cache.match("https://unpkg.com/@vimeo/player@2.20.1/dist/player.es.js")
```

but now it throws the following exception:

```
err: NotFound: No such file or directory (os error 2)
at async Cache.[[[matchAll]]] (ext:deno_cache/01_cache.js:243:27)
at async Cache.match (ext:deno_cache/01_cache.js:167:15)
...
name: "NotFound",
code: "ENOENT"
```

After clear the cache or put again the http response, it works fine.
I don't know how to reproduce this error, maybe it happens after some days when the cache become stale.

**Comment by bartlomieju** [comment]
I believe the problem is here: 
https://github.com/denoland/deno/blob/576b20aa005c233b95afe6e3692a899ae8f755f2/ext/cache/sqlite.rs#L264

If the file doesn't exist we should error out but instead return empty response and clear relevant cache entry in the DB. Not sure how to test it though.

**Comment by mmastrac** [comment]
Fixed in #21949, though this is a challenge to add a test for.
